### PR TITLE
Drop Python 3.8 Support

### PR DESCRIPTION
**Pull request by chezou**
## Description

Since Python 3.8 is EoL, this change makes Python 3.12 installation easier, especially by preventing pandas build since locked pandas version does not have wheels for Python 3.12.

- Drop Python 3.8
- Update RTD Python version

PyTorch will likely support Python 3.13 early next year, so I skipped to support Python 3.13.
https://github.com/pytorch/pytorch/issues/130249

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Optimization

## How Has This Been Tested?
Before submitting a PR, please check yourself against the following list. It would save us quite a lot of time.
- Have you read the contribution guide?: Yes
- Have you updated the relevant docstrings? We're using Numpy format, please double-check yourself: NA
- Does your change require any new tests?: NA
- Have you updated the changelog file?

### CODE REVIEW

**Comment by feldlime** [comment]
@chezou Thanks for the PR!

Before merging, we have to ensure that nobody in the company uses Python 3.8. It may take a few days

**Comment by chezou** [comment]
@feldlime Got it.

Due to the PyTorch issue, the current pyproject.toml fails to install with Python 3.8 https://github.com/pytorch/pytorch/issues/139005

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `load_model` function ([#213](https://github.com/MobileTeleSystems/RecTools/pull/213))
 - `model_from_config` function ([#214](https://github.com/MobileTeleSystems/RecTools/pull/214))
 - `get_cat_features` method to `SparseFeatures` ([#221](https://github.com/MobileTeleSystems/RecTools/pull/221))
+- Drop Python 3.8 Support ([#222](https://github.com/MobileTeleSystems/RecTools/pull/222))
```
Comment (feldlime) [inline CHANGELOG.md:22]:
```suggestion

### Removed
- Python 3.8 support ([#222](https://github.com/MobileTeleSystems/RecTools/pull/222))
```

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -33,7 +33,7 @@ Pull Request Process
 #. Create a virtual environment and install dependencies including all 
    extras and development dependencies.
    
-   #. Make sure you have ``python>=3.8`` and ``poetry>=1.5.0`` installed
+   #. Make sure you have ``python>=3.9`` and ``poetry>=1.5.0`` installed
```
Comment (feldlime) [inline CONTRIBUTING.rst:36]: Please also fix this in the `security.md`

**Review by feldlime — CHANGES_REQUESTED** [review]
We checked and can confirm that we can drop 3.8 support safely

Just 2 small comments + you need to pull the fresh version of the `main` branch

File: CONTRIBUTING.rst (hunk 2 above)
Comment (chezou) [reply CONTRIBUTING.rst:36]: Thanks, fixed 
b0b170f

**Comment by chezou** [comment]
@feldlime Applied review comments and rebased toward main.

**Review by feldlime — APPROVED** [review]

**Pull request merged**